### PR TITLE
WIP: Rhmap 17324

### PR DESCRIPTION
**Pull request by pmacko1**
implementing comments from older gdoc, i hope that the current repo does not differ much. https://docs.google.com/document/d/11DNIf2pFG6KcXrMwWXlHz3QnzZpcnifcUTQBzWv_Nj8/edit?ts=59706905# 
Please only review, I will merge it afterwards.

29 aug: + comments from https://github.com/aerogear/aerogear.org/pull/659/

### CODE REVIEW

File: docs/digger/installation/ansible-installs-jenkins.asciidoc [hunk 1]
```diff
@@ -1,9 +1,10 @@
-=== How Ansible Installs Jenkins
+=== Upload User Public Keys for Automatic Jenkins Installation
 //pwright: I think heading should be task oriented, ie run playbook
 
-The installer will perform the following tasks for Jenkins setup and configuration:
+The installer creates the Jenkins service on OpenShift with required routes, services and pods, and the waits for Jenkins to be running. It also creates an admin user.
+//Admin user where ? ^
```
Comment (pmacko1) [inline docs/digger/installation/ansible-installs-jenkins.asciidoc]: pls resolve

File: docs/digger/installation/ansible-installs-jenkins.asciidoc [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 
-* Create the Jenkins service on OpenShift. It will automatically create the required routes, services and pods, and wait for Jenkins to be up and running. It will also create an admin user by default.
-* Then the installer will use the https://wiki.jenkins.io/display/JENKINS/Jenkins+CLI[Jenkins CLI] tool to configure Jenkins. This tool requires the user public keys to be uploaded to Jenkins manually. The installer will stop and provide you with instructions on how to do that.
-* Once the key is uploaded, the installer will resume. It will update configurations of Jenkins, and install the required plugins.
+Upload the User Public keys to Jenkins with steps printed in the terminal when prompted by the Installer. These keys will be used by Ansible to Jenkins configuration by Jenkins CLI tool.
+//Are the configurations AND plugins installed by Jenkins CLI tool? ^ If j then the last sentence is not required and "installation of required plugins" goes before "by Jenkins CLI tool".
```
Comment (pmacko1) [inline docs/digger/installation/ansible-installs-jenkins.asciidoc]: pls resolve

File: docs/digger/installation/ansible-installs-jenkins.asciidoc [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
 
+The installation will continue when the key are uploaded. It will update the configurations of Jenkins, and install the required Jenkins plugins.
```
Comment (pmacko1) [inline docs/digger/installation/ansible-installs-jenkins.asciidoc]: pls verify

File: docs/digger/installation/digger-prereqs.asciidoc [hunk 4]
```diff
@@ -11,12 +11,12 @@ Please check http://docs.ansible.com/ansible/intro_installation.html[Ansible ins
 
 === Java 2.2 and higher
 
-A local installation of Java is required.
+A local installation of Java is required. DOES THIS NEED JDK???? WHICH VERSION?
 //does this need JDK? what versions are supported?"
```
Comment (pmacko1) [inline docs/digger/installation/digger-prereqs.asciidoc]: pls resolve

File: docs/digger/installation/install-android-sdk.asciidoc [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
 * It doesn't require a build server to be running constantly
 * Each build is performed completely in isolation
 
-The Android SDK is required to build Android apps. The installer will download the Android SDK and make it available to the building pods by performing the following tasks:
-
-* Creating a new pod which will be used to download the Android SDK. The pod will use an image that contains the required tools, and claim a Persistent Volume(PV).
-* Invoke a script to download the Android SDK. During the process, the user will be prompted to accept the Android SDK License Agreement. The Android SDK will be persisted in the PV.
-* Update the Kubernetes plugin configuration to add a new pod template. It will mount the PV in “Read-Only” mode to the pods and setup the required environment variables to define an Android home directory.
 
 ==== Configure the Android SDK components
```
Comment (pmacko1) [inline docs/digger/installation/install-android-sdk.asciidoc]: remove 'the'

File: docs/digger/installation/ansible-installs-jenkins.asciidoc (hunk 3 above)
Comment (PhilipGough) [reply docs/digger/installation/ansible-installs-jenkins.asciidoc]: This is correct

File: docs/digger/installation/ansible-installs-jenkins.asciidoc (hunk 2 above)
Comment (PhilipGough) [reply docs/digger/installation/ansible-installs-jenkins.asciidoc]: Yes they are installed by jenkins cli

File: docs/digger/installation/ansible-installs-jenkins.asciidoc (hunk 1 above)
Comment (PhilipGough) [reply docs/digger/installation/ansible-installs-jenkins.asciidoc]: I dont think this is accurate. The installer creates the Jenkins instance and route but I don't believe we create an admin user ping @odra for clarification

Comment (odra) [reply docs/digger/installation/ansible-installs-jenkins.asciidoc]: Correct, the installer assumes that the admin/jenkins user was already created with, we have a sample bash script that creates this user: [https://github.com/aerogear/digger-jenkins/blob/master/admin/create-osx-user.sh](https://github.com/aerogear/digger-jenkins/blob/master/admin/create-osx-user.sh)

Note: The mac machine may need to be rebooted after creating this user.

**Comment by laurafitzgerald** [comment]
@PhilipGough could you take a quick look here. I have removed the comments to merge. There are some changes in the macos-prereqs which will probably be out of date given your pr.

File: docs/digger/installation/ansible-installs-jenkins.asciidoc (hunk 2 above)
Comment (PhilipGough) [reply docs/digger/installation/ansible-installs-jenkins.asciidoc]: @laurafitzgerald still some comments here

File: docs/digger/installation/digger-install-intro.asciidoc [hunk 6]
```diff
@@ -1,16 +1,16 @@
 == Digger Installation Introduction
 //Please DO NOT use 'Overview' as a section heading anywhere http://stylepedia.net/#sect-Red_Hat_Technical_Publications-Writing_Style_Guide-Overall_Book_Design-Unused_Heading_Titles
+GitHub repo with Ansible job used for Digger installation https://github.com/aerogear/digger-installer[Digger Installer tool].
 
-The AeroGear Digger project allows you to setup a service builds mobile applications using OpenShift containers.
-//DON'T USE PASSIVE VOICE
+Use AeroGear Digger to building mobile applications on OpenShift containers.
```
Comment (PhilipGough) [inline docs/digger/installation/digger-install-intro.asciidoc]: to --> for

File: docs/digger/installation/digger-install-intro.asciidoc [hunk 7]
```diff
@@ -1,16 +1,16 @@
...
 * macOS servers with pre-installed tools to build iOS applications
-* An http://docs.ansible.com/ansible/index.html[Ansible] based automation tool used to install and configure all the components
-* A suite of sample Jenkinsfiles to demonstrate how to build mobile applications using Jenkins pipeline
+* An http://docs.ansible.com/ansible/index.html[Ansible] based automation tool used to install and configure all the required components
+* A suite of sample `Jenkinsfiles` to demonstrate how to build mobile applications using Jenkins pipeline
 
-The main objective of this project is to be used with the RedHat Application Platform (RHMAP).
-However, it will be possible to use it on its own as a CI/CD server for mobile applications.
+This project is designed to be used with the RedHat Application Platform (RHMAP). 
+However, it is possible to use it on its own as a CI/CD server for mobile applications.
```
Comment (PhilipGough) [inline docs/digger/installation/digger-install-intro.asciidoc]: for --> to build?

File: docs/digger/installation/digger-prereqs.asciidoc [hunk 8]
```diff
@@ -1,22 +1,13 @@
...
-
-A local installation of Java is required.
-//does this need JDK? what versions are supported?"
-
-=== OpenShift 3.0 and higher
+=== OpenShift 3.3 and higher
 
 * A running OpenShift cluster, and its command line tool (https://github.com/openshift/origin/releases/tag/v1.3.1[oc]) installed locally.
-* The OpenShift cluster should have at least 2 persistent volumes (PVs) with 10G capacity or more.
+* The OpenShift cluster must have at least 2 persistent volumes (PVs) with minimum of 10GB instantiated.
```
Comment (PhilipGough) [inline docs/digger/installation/digger-prereqs.asciidoc]: is this correct, i think we need 3 and we need a 40?

File: docs/digger/installation/install-android-sdk.asciidoc [hunk 9]
```diff
@@ -1,21 +1,23 @@
-=== How Installer installs Android SDK
+=== How to install Android SDK with Ansible
 //pwright: I think heading should be task oriented, ie run playbook
+//pmacko: +1, changed to active form
```
Comment (PhilipGough) [inline docs/digger/installation/install-android-sdk.asciidoc]: ?

File: docs/digger/installation/install-android-sdk.asciidoc [hunk 10]
```diff
@@ -1,21 +1,23 @@
...
+WARNING: During the process, you will be ask to manually accept the Android SDK License Agreement.
```
Comment (PhilipGough) [inline docs/digger/installation/install-android-sdk.asciidoc]: ask --> asked

**Comment by laurafitzgerald** [comment]
@PhilipGough changes made as suggested. further comments removed.

File: docs/digger/installation/install-android-sdk.asciidoc [hunk 11]
```diff
@@ -1,21 +1,23 @@
...
+* Invoke a script to download the Android SDK. The Android SDK will be retained in the PV.
+* Update the Kubernetes plugin configuration to add a new pod template. It will mount and connect the PV in “Read-Only” mode to the pods, and setup the required environment variables to define an Android home directory.
 
-* It doesn't require a build server to be running constantly
-* Each build is performed completely in isolation
+Jenkins dynamically creates containers (pods) on OpenShift using the https://wiki.jenkins.io/display/JENKINS/Kubernetes+Plugin[Kubernetes Jenkins plugin] for each Android build.
 
-The Android SDK is required to build Android apps. The installer will download the Android SDK and make it available to the building pods by performing the following tasks:
+//* It doesn't require a build server to be running constantly
+//* Each build is performed completely in isolation -> redundant I think, good for "marketing", otherwise I think not, ppl are familiar with containers
```
Comment (PhilipGough) [inline docs/digger/installation/install-android-sdk.asciidoc]: ?

File: docs/digger/installation/install-android-sdk.asciidoc [hunk 12]
```diff
@@ -1,21 +1,23 @@
...
-* Update the Kubernetes plugin configuration to add a new pod template. It will mount the PV in “Read-Only” mode to the pods and setup the required environment variables to define an Android home directory.
 
-==== Configure the Android SDK components 
+==== Configure Android SDK components 
 
-The Android SDK contains many different components. 
-Only a subset of them are installed by default.
-The list of components that will be installed are specified https://github.com/aerogear/digger-installer/blob/master/android-sdk/templates/sample_cfg.j2[in this file] contained in digger-installer repo .
+The Android SDK contains many different components. Only a subset of them are installed by default.
+The list of installed components are specified https://github.com/aerogear/digger-installer/blob/master/android-sdk/templates/sample_cfg.j2[in this file] contained in `digger-installer` repo .
```
Comment (PhilipGough) [inline docs/digger/installation/install-android-sdk.asciidoc]: Dont think pointing them at the jinja file is a good idea, best to refer to vars, we do not want them editing this file

File: docs/digger/installation/digger-prereqs.asciidoc (hunk 8 above)
Comment (laurafitzgerald) [reply docs/digger/installation/digger-prereqs.asciidoc]: this is resolved in my pr

File: docs/digger/installation/install-android-sdk.asciidoc [hunk 13]
```diff
@@ -1,21 +1,23 @@
...
 
-You can modify this file to select the Android SDK components for installation.
\ No newline at end of file
+You can modify this file to select which Android SDK components you want to install.
```
Comment (PhilipGough) [inline docs/digger/installation/install-android-sdk.asciidoc]: This is a big no no

File: docs/digger/installation/install-android-sdk.asciidoc (hunk 12 above)
Comment (laurafitzgerald) [reply docs/digger/installation/install-android-sdk.asciidoc]: yes agreed, this was commented on in the google doc. will resolve

File: docs/digger/installation/macos-prereqs.asciidoc [hunk 14]
```diff
@@ -1,63 +1,63 @@
...
+. Set a secure user `PASSWORD` in `create-osx-user.sh`.
+. Run `create-osx-user.sh`:
 +
 ----
 sudo bash create-osx-user.sh
 ----
 +
-. Execute `enable-osx-remote-login.sh` to allow SSH access:
+. Run `enable-osx-remote-login.sh` to enable SSH access:
 +
```
Comment (PhilipGough) [inline docs/digger/installation/macos-prereqs.asciidoc]: I think most of the above is covered in the new pr

File: docs/digger/installation/install-android-sdk.asciidoc (hunk 13 above)
Comment (laurafitzgerald) [reply docs/digger/installation/install-android-sdk.asciidoc]: as above

File: docs/digger/installation/macos-prereqs.asciidoc [hunk 15]
```diff
@@ -1,63 +1,63 @@
...
-| To download and install Homebrew packages and Cocoapods.
+| `Homebrew packages` and `Cocoapods`
 
 | https://raw.githubusercontent.com
-| To download and install NVM.
+| `NVM - Node Version Manager`
 
 | https://rvm.io
-| To download and install RVM.
+| `RVM - Ruby Version Manager`
```
Comment (PhilipGough) [inline docs/digger/installation/macos-prereqs.asciidoc]: This isnt valid

File: docs/digger/installation/rhmap-properties.asciidoc [hunk 16]
```diff
@@ -21,40 +21,37 @@ using Digger.
...
 
 | buildfarm_build_polling_interval
 | BUILDFARM_BUILD_POLLING_INTERVAL
 | The number of seconds between each poll of a Jenkins build.
 
 | buildfarm_csrf_enabled
 | BUILDFARM_CSRF_ENABLED
-| Whether CSRF protection is enabled on the Jenkins instance. It is recommended
-to enable CSRF protection.
+| CSRF protection. Recommended setting: Enabled
```
Comment (PhilipGough) [inline docs/digger/installation/rhmap-properties.asciidoc]: are the optinos not `true,false`

File: docs/digger/installation/macos-prereqs.asciidoc (hunk 15 above)
Comment (laurafitzgerald) [reply docs/digger/installation/macos-prereqs.asciidoc]: is this removed in your pr?

File: docs/digger/suggestions [hunk 17]
```diff
@@ -24,4 +24,6 @@ HTTP Proxy Usage Limitations: Can we refer to some other documentation regarding
 - should widely used terms such as:
   .Ansible Playbook
   .Jenkins Plugin
-  etc. be CAPITALISED?
\ No newline at end of file
+  etc. be CAPITALISED?
+  
+- don't use "machine", use "system" instead. Machine sounds very "mechanical". ????
```
Comment (PhilipGough) [inline docs/digger/suggestions]: ?

File: docs/digger/installation/rhmap-properties.asciidoc (hunk 16 above)
Comment (laurafitzgerald) [reply docs/digger/installation/rhmap-properties.asciidoc]: I have made a note of it with the other comments from this pr. will resolve in my pr

File: docs/digger/installation/macos-prereqs.asciidoc (hunk 14 above)
Comment (laurafitzgerald) [reply docs/digger/installation/macos-prereqs.asciidoc]: :+1

File: docs/digger/suggestions (hunk 17 above)
Comment (laurafitzgerald) [reply docs/digger/suggestions]: @PhilipGough  I believe this file is being used by @pmacko1 to help with writing the documentation. I don't want to remove it altogether but I think it should be added to .gitignore by @pmacko1 so he can keep it locally.

**Comment by laurafitzgerald** [comment]
@PhilipGough shall I merge this one. All comments are resolved here or to be resolve in your pr and my pr.

**Pull request merged**